config/application.rb

```diff
@@ -76,10 +76,10 @@ class Application < Rails::Application
       Spree::Config['checkout_zone'] = ENV['CHECKOUT_ZONE']
       Spree::Config['currency'] = ENV['CURRENCY']
 
-      if ActiveRecord::Base.connected? && Spree::Country.table_exists?
+      begin
         country = Spree::Country.find_by(iso: ENV['DEFAULT_COUNTRY_CODE'])
         Spree::Config['default_country_id'] = country.id if country.present?
-      else
+      rescue ::ActiveRecord::StatementInvalid
         Spree::Config['default_country_id'] = 12  # Australia
```

Comment:
So... if anything ever goes wrong here, we'll set the server's default country to Australia, even in production? But actually those IDs can be different depending on an instances configuration, so a country with an ID of`12` might not even exist?

That makes me a little bit uneasy...

Also; we're rescuing `ActiveRecord::StatementInvalid` instead of `ActiveRecord::NoDatabaseError`?

Reply:
Just looking at all the other places where we rely on `Spree::Config[:default_country_id]` in general, it seems like it's not a good idea in the first place... The number of problems we've ad over the years related to the weird caching that happens with values in `Spree::Config` has been so huge (and is still ongoing, eg: mail configs).

I'm wondering if we should just start trying to move away from dependence on `Spree::Config` a bit?

Like in `EnterpriseGroupsController#build_resource` instead of doing this: 
```ruby
enterprise_group.address.country = Spree::Country.find_by(
  id: Spree::Config[:default_country_id]
)
```
we could just do this:
```ruby
enterprise_group.address.country = Spree::Country.find_by(
  iso: ENV["DEFAULT_COUNTRY_CODE"]
)
```

What do you think?

Reply:
If that's good with you it's good with me. At first glance I like that a lot better. I'm trying to think if there's some downside to relying on an ENV variable like that over the `Spree::Config` wrapper and I can't immediately think of one.